plugins/xterm/xterm.plugin.bash

```diff
@@ -12,23 +12,23 @@ _omb_module_require plugin:bash-preexec
 
 function _omb_plugin_xterm_short_dirname {
   local dir_name=${PWD/~/\~}
-  if [[ ${OMB_PLUGIN_XTERM_SHORT_TERM_LINE-} == true ]] && ((${#dir_name} > 8)); then
+  if [[ ${OMB_PLUGIN_XTERM_SHORT_TERM_LINE:-} == true ]] && ((${#dir_name} > 8)); then
     dir_name=${dir_name##*/}
   fi
   echo "$dir_name"
 }
 
 function _omb_plugin_xterm_short_command {
   local input_command="$*"
-  if [[ ${OMB_PLUGIN_XTERM_SHORT_TERM_LINE-} == true ]] ((${#input_command} > 8)); then
+  if [[ ${OMB_PLUGIN_XTERM_SHORT_TERM_LINE:-} == true ]] && ((${#input_command} > 8)); then
     input_command=${input_command%% *}
   fi
   echo "$input_command"
 }
 
 function _omb_plugin_xterm_set_title {
-  local title=${1-}
-  printf '\e]0;%s\e\\' "$title"
+  local title=${1:-xterm}
+  printf '\e]0;%s\a' "$title"
```

Comment:
ANSI standard requires an OSC sequence to be terminated by ST (`\e\\`) but not by BEL (`\a`). The convention of the OSC sequence terminated by BEL is derived from the historical behavior that XTerm treats *any* control character as the terminator of the OSC sequence, but this is undefined in the standard. Terminal WG, the community of terminal developers, is moving toward deprecating BEL and unifying all the OSC sequences to be terminated by ST [1] (which is a part of Merge Request [2]).

- [1] [Title Setting - index.md · 45c1c6c4 · terminal-wg / specifications · GitLab](https://gitlab.freedesktop.org/terminal-wg/specifications/-/blob/45c1c6c46e0de498064e826f63e1daf3ab400f82/proposals/feature-reporting/index.md#title-setting)
- [2] [Feature Reporting. (!8) · Merge requests · terminal-wg / specifications · GitLab](https://gitlab.freedesktop.org/terminal-wg/specifications/-/merge_requests/8/diffs#52c1a05f148e1d116aa84a794c9f37ff44b85ec4)

Are there any real issues with ST (`\e\\`)? If not, could you revert it to use ST (`\e\\`)?

Reply:
Wrt. to the sequence: sure, I will. I learned something new.
As for 'title', the change means set `xterm` as the name when `$1` is empty. Is this OK?

Reply:
> As for 'title', the change means set `xterm` as the name when `$1` is empty. Is this OK?

Thanks for your reply. The change to `title` is OK. Please keep it and change only `\a`.

Reply:
> I learned something new.

FYI, you may find Ref. [3] interesting. This is a part of a Merge Request for [Terminal WG/terminal-parsing](https://gitlab.freedesktop.org/terminal-wg/terminal-parsing), and this section describes the historical background of the OSC sequences.

- [3] [4.3 OSC sequence (DEC/XTerm extension) - doc/n0001.md · n0001r0 · terminal-parsing · GitLab](https://gitlab.freedesktop.org/akinomyoga/terminal-parsing/-/blob/n0001r0/doc/n0001.md#43-osc-sequence-decxterm-extension)